### PR TITLE
[containers] fix CPU metric reporting when cgroup file is missing

### PR DESCRIPTION
**Pull request by celenechang**
### What does this PR do?

Thanks to the detailed investigation in https://github.com/DataDog/datadog-agent/issues/7370, it seems that cgroup files can occasionally and briefly be missing when gathering resource data. For CPU in particular it leads to grossly misleading metrics since the data are reported cumulatively. This PR changes how we handle the case of a missing cgroup file when gathering CPU data - instead of returning an empty struct or `nil` (which would eventually be replaced with an empty struct), which results in reporting 0 values, we return an error and don't report anything.

### Motivation

Better user experience (and reduced false positives)

### Additional Notes

Anything else we should know when reviewing?

### Describe how to test your changes

Ideally, run the agent in a Docker container for a few days and make sure `docker.cpu.usage` metric does not spike (e.g. over 1k%)

### CODE REVIEW

File: pkg/util/containers/providers/cgroup/cgroup_metrics.go [hunk 1]
```diff
@@ -182,17 +182,20 @@ func (c ContainerCgroup) SoftMemLimit() (uint64, error) {
...
+// struct will be defaulted and set CPU values to 0, leading to misleading results
+// (since CPU is reported cumulatively).
 func (c ContainerCgroup) CPU() (*metrics.ContainerCPUStats, error) {
-	ret := &metrics.ContainerCPUStats{}
 	statfile := c.cgroupFilePath("cpuacct", "cpuacct.stat")
 	f, err := os.Open(statfile)
 	if os.IsNotExist(err) {
 		log.Debugf("Missing cgroup file: %s", statfile)
-		return ret, nil
+		return nil, fmt.Errorf("cgroup file does not exist")
```
Comment (vboulineau) [inline pkg/util/containers/providers/cgroup/cgroup_metrics.go:193]: I think we can remove the debug log and put the `statfile` inside the error

Comment (celenechang) [reply pkg/util/containers/providers/cgroup/cgroup_metrics.go:193]: thanks! i was thinking the same thing, but then thought i'd leave it for consistency with the other methods. think it's too redundant? i don't feel strongly either way

Comment (vboulineau) [reply pkg/util/containers/providers/cgroup/cgroup_metrics.go:193]: We can keep the `Debug` log for consistency but in any case the error should container the file path, it'll ease investigations when we get non-debug flares.

Comment (celenechang) [reply pkg/util/containers/providers/cgroup/cgroup_metrics.go:193]: makes sense!

**Review by vboulineau — APPROVED** [review]

**Pull request merged**